src/sentry/api/event_search.py

```diff
@@ -94,45 +94,55 @@
 aggregate_rel_date_filter = negation? aggregate_key sep operator? rel_date_format
 
 # has filter for not null type checks
-has_filter = negation? "has" sep (search_key / search_value)
+has_filter = negation? &"has:" search_key sep (search_key / search_value)
```

Comment:
This is marginally nicer on the frontend since the token still comes out as a search key. We just use the lookahead assertion to make sure the key is `has:`